### PR TITLE
More CUDAImageEncoder (now CUDAJPEGProcessor) Improvements

### PR DESCRIPTION
**Pull request by TomaszTB**
This PR:
* simplifies method calls to encode images
* adds method to encode gray scale image
* adds decoding ability

### CODE REVIEW

File: ihmc-high-level-behaviors/src/libgdx/java/us/ihmc/rdx/ui/graphics/ros2/pointCloud/RDXROS2ColoredPointCloudVisualizerColorChannel.java
Comment (TomaszTB) [inline ihmc-high-level-behaviors/src/libgdx/java/us/ihmc/rdx/ui/graphics/ros2/pointCloud/RDXROS2ColoredPointCloudVisualizerColorChannel.java:1]: This seems like the perfect class to use the CUDA jpeg decoding. Let me know if there's any other places we should use it.

Comment (ds58) [reply ihmc-high-level-behaviors/src/libgdx/java/us/ihmc/rdx/ui/graphics/ros2/pointCloud/RDXROS2ColoredPointCloudVisualizerColorChannel.java:1]: Is CUDA a requirement of RDX already? Should we expect anyone running the UI to need a CUDA device?

**Review by ds58 — APPROVED** [review]

File: ihmc-high-level-behaviors/src/libgdx/java/us/ihmc/rdx/ui/graphics/ros2/pointCloud/RDXROS2ColoredPointCloudVisualizerColorChannel.java
Comment (TomaszTB) [reply ihmc-high-level-behaviors/src/libgdx/java/us/ihmc/rdx/ui/graphics/ros2/pointCloud/RDXROS2ColoredPointCloudVisualizerColorChannel.java:1]: Oh that's a great point. I don't recall RDX requiring CUDA as of yet. Perhaps I should remove the RDX changes from this PR?

Comment (TomaszTB) [reply ihmc-high-level-behaviors/src/libgdx/java/us/ihmc/rdx/ui/graphics/ros2/pointCloud/RDXROS2ColoredPointCloudVisualizerColorChannel.java:1]: Ultimately I'd like to have something like a generic `ImageProcessor` which internally checks whether CUDA is available or not, and defaults to OpenCV if no CUDA devices are found.

Comment (ds58) [reply ihmc-high-level-behaviors/src/libgdx/java/us/ihmc/rdx/ui/graphics/ros2/pointCloud/RDXROS2ColoredPointCloudVisualizerColorChannel.java:1]: Perhaps we should do that so the UI stuff doesn't have to require CUDA

**Pull request merged**